bpf/lib/nat.h

```diff
@@ -338,9 +339,10 @@ static __always_inline int snat_v4_track_connection(struct __ctx_buff *ctx,
 	memcpy(&tmp, tuple, sizeof(tmp));
 
 	where = dir == NAT_DIR_INGRESS ? CT_INGRESS : CT_EGRESS;
+	ipv4_ct_tuple_swap_ports(&tmp);
 
-	ret = ct_lookup4(get_ct_map4(&tmp), &tmp, ctx, off, where,
-			 &ct_state, &monitor);
+	ret = ct_lazy_lookup4(get_ct_map4(&tmp), &tmp, ctx, off, has_l4_header,
+			      ct_action, where, &ct_state, &monitor);
```

Comment:
Mind splitting `bpf: Reuse the NAT tuple for ct_lookup4` into two patches - one trivial refactor that only does the rename & introduces the `action` parameter, and a second that actually adapts the IPv4 NAT code to use `ct_lazy_lookup4()` ?

I think that would make it a lot easier to follow what actually changed.